sauron/tests/test_sauron_esplora_bitcoin.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/python
+
+import os
+
+import pyln
+import pytest
+from pyln.testing import utils
+from pyln.testing.fixtures import *  # noqa: F403
+from util import LightningD
+
+pyln.testing.fixtures.network_daemons["bitcoin"] = utils.BitcoinD
+
+
+class LightningNode(utils.LightningNode):
+    def __init__(self, *args, **kwargs):
+        pyln.testing.utils.TEST_NETWORK = "bitcoin"
+        utils.LightningNode.__init__(self, *args, **kwargs)
+        lightning_dir = args[1]
+
+        self.daemon = LightningD(lightning_dir, None)  # noqa: F405
+        options = {
+            "disable-plugin": "bcli",
```

Comment:
I don't think you need to override this class at all. Just set the options appropriately when using the `NodeFactory.get_node()`. That'll obviate the need for the subclass and the fixtures.

Reply:
Hey @cdecker it looks like we actually do need to override both the `LightningNode` class and the `LightningD` class. 

We need to override the `LightningNode` class because it doesn't allow us to override `self.daemon` as you can see in this code here: https://github.com/ElementsProject/lightning/blob/e66653fa1d847e28bd812356a762225a44bfae83/contrib/pyln-testing/pyln/testing/utils.py#L780-L783

In addition, we have to override the `LightningD` class because there is no way to instantiate it without the `bcli` options which are set here: https://github.com/ElementsProject/lightning/blob/e66653fa1d847e28bd812356a762225a44bfae83/contrib/pyln-testing/pyln/testing/utils.py#L618-L622

These options if set, crash `lightningd` when `bcli` is disabled.

On top of this, the `LightningD.start`method requires `bitcoin-rpcuser` to be set (which as above crashes `lightningd` when we disable `bcli`). 

So unfortunately we need to leave these classes as they are unless I'm missing something.